src/java.base/share/classes/java/text/MessageFormat.java

```diff
@@ -110,17 +110,21 @@
  * </pre></blockquote>
  *
  * <p>
- * The <i>ArgumentIndex</i> value is a non-negative integer written
+ * The <b>ArgumentIndex</b> value is a non-negative integer written
  * using the digits {@code '0'} through {@code '9'}, and represents an index into the
  * {@code arguments} array passed to the {@code format} methods
  * or the result array returned by the {@code parse} methods.
  * <p>
- * The <i>FormatType</i> and <i>FormatStyle</i> values are used to create
+ * Any constructor or method that takes a String pattern parameter will throw an {@code IllegalArgumentException} if the
+ * pattern contains an <b>ArgumentIndex</b> value that is equal to or exceeds an implementation limit.
+ * <p>
+ * The <b>FormatType</b> and <b>FormatStyle</b> values are used to create
  * a {@code Format} instance for the format element. The following
  * table shows how the values map to {@code Format} instances. These values
  * are case-insensitive when passed to {@link #applyPattern(String)}. Combinations
  * not shown in the table are illegal. A <i>SubformatPattern</i> must
  * be a valid pattern string for the {@code Format} subclass used.
+ * @implSpec For this implementation, the limit of <b>ArgumentIndex</b> is 10,000
```

Comment:
I don't think the number should be *specified*; it would be fine as an implementation note.